python-threatexchange/threatexchange/extensions/vpdq/vpdq_index.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Implementation of SignalTypeIndex abstraction for PDQ by wrapping
+hashing.pdq_faiss_matcher.
+"""
+
+import typing as t
+import vpdq
+from threatexchange.signal_type.index import (
+    SignalTypeIndex,
+    VPDQIndexMatch,
+    IndexMatch,
+    T as IndexT,
+)
+from threatexchange.extensions.vpdq.vpdq_faiss import VPDQHashIndex, VPDQFlatHashIndex
+from threatexchange.extensions.vpdq.vpdq_util import (
+    prepare_vpdq_feature,
+    VPDQMatchResult,
+    VPDQ_CONFIDENT_QUALITY_THRESHOLD,
+    VPDQ_CONFIDENT_DISTANCE_THRESHOLD,
+)
+
+
+class VPDQIndex(SignalTypeIndex):
+    """
+    Wrapper around the vpdq faiss index lib using VPDQFlatHashIndex
+    """
+
+    @classmethod
+    def get_match_distance_threshold(cls) -> int:
+        return VPDQ_CONFIDENT_DISTANCE_THRESHOLD
+
+    @classmethod
+    def _get_empty_index(cls) -> VPDQHashIndex:
+        return VPDQFlatHashIndex()
+
+    def __init__(
+        self,
+        entries: t.Iterable[t.Tuple[str, IndexT]] = (),
+        quality_threshold: int = VPDQ_CONFIDENT_QUALITY_THRESHOLD,
+    ) -> None:
+        super().__init__()
+        self.index: VPDQHashIndex = self._get_empty_index()
+        self.entries: t.List = []
+        self.idx_to_vpdq: t.List[t.Tuple[int, vpdq.VpdqFeature]] = []
```

Comment:
This name surprised me a little bit - my best guess for alternatives:
1. faiss_id_to_entry_id
2. index_idx_to_entry_idx
(or some variation)

Also, I'm not 100% you actually need the original VpdqFeature anywhere that you would need this unless you plan on verifying with the brute force. You might be doubling your storage cost without too much benefit, but I don't know if you are relying on this for testing.